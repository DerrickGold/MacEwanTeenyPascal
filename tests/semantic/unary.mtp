var something : integer;

begin

	something := -2 * 3;
end.