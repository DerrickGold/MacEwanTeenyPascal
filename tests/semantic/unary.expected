
Semantically Corrected AST
[Head] (Block, Block Statement) 
Symbol Table:============================================================================:
Identifiers    |Stack Offset  |Type                 |Value                           
-----------------------------------------------------------------------------------------
something      |0             |Int, Var             |                                
=======================================================================Depth 0   :Size 4   
    [Child] (Variable Section) 
        [Child] (Variable Declaration, Variable List) 
            [Child] 1, TOK_ID, something, (Literal) 
            [Child] 1, TOK_KEY_INTEGER, integer, (Type, Integer) 
    [Child] (Statement List) 
        [Child] (Assign Statement) 
            [Child] 5, TOK_ID, something, (Variable) 
            [Child] 5, TOK_MINUS, -, (Unary Operator) 
                [Child] 5, TOK_STAR, *, (Multiply Operator, Term) 
                    [Child] 5, TOK_NUM, 2, (Constant, Integer) 
                    [Child] 5, TOK_NUM, 3, (Constant, Integer) 
        [Sibling] (Null Statement) 
